python/pyspark/sql/functions.py

```diff
@@ -85,13 +96,16 @@ def _():
     >>> df.select(lit(5).alias('height')).withColumn('spark_user', lit(True)).take(1)
     [Row(height=5, spark_user=True)]
     """
-_functions = {
+_name_functions = {
+    # name functions take a column name as their argument
     'lit': _lit_doc,
```

Comment:
Does `lit` takes the string as column names? how do we create string literal?

Reply:
and .. what's really "name function" ... ?

Reply:
No, `lit()` takes a literal value and creates a column with that literal. That's what it's for. It doesn't make sense for it to accept a column name, given its nature. So if you give it a string it will create a column with that string literal.

The name "name function" is something I came up with just to distinguish these functions from the ones that take columns as input. They are defined by that distinction - they are "functions that take a column name as their argument", exclusively.

Reply:
`lit` doesn't take a column name as their argument. Why did we use such `name function` category for `lit`?

Reply:
To be fair, `lit` is such a unique function semantically that it would warrant its own category. But since the implementation is exactly the same as those "name functions", I left it there for practical purposes.

Reply:
One (lit) of five (col, column, asc, desc, lit) doesn't sound like a special case tho. It had to have a better category if 20% of items doesn't fit to the category.